Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -161,6 +249,12 @@ BTDiagnostics::DefineCellCenteredMultiFab(int lev)
     // This MultiFab will store all the user-requested fields in the boosted-frame
     auto & warpx = WarpX::GetInstance();
     // The BoxArray is coarsened based on the user-defined coarsening ratio
+    amrex::BoxArray ba = warpx.boxArray(lev);
```

Comment:
Do we need to manipulate this as reference?

Reply:
We are modifying this `ba` below so we don't want it to be a reference.

`BoxArray` is actually pretty cheap to copy bc it doesn't do a deep copy. The reason is that if you modify a BoxArray in a simple way, e.g. by coarsening, you don't want to create a whole new dense representation, you just remember "I'm like the one, but coarsened by some factor". So I think this is actually an efficient way to do this.

Reply:
Thank you for the details! :)